addons/io/org.openhab.io.homekit/src/main/java/org/openhab/io/homekit/internal/HomekitImpl.java

```diff
@@ -0,0 +1,90 @@
+package org.openhab.io.homekit.internal;
+
+import java.io.IOException;
+import java.net.UnknownHostException;
+import java.security.InvalidAlgorithmParameterException;
+
+import org.eclipse.smarthome.core.items.ItemRegistry;
+import org.eclipse.smarthome.core.storage.StorageService;
+import org.openhab.io.homekit.Homekit;
+import org.osgi.service.component.ComponentContext;
+import org.osgi.service.component.annotations.Activate;
+import org.osgi.service.component.annotations.Deactivate;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.beowulfe.hap.HomekitRoot;
+import com.beowulfe.hap.HomekitServer;
+
+/**
+ * Provides access to OpenHAB items via the Homekit API
+ *
+ * @author Andy Lintner
+ */
+public class HomekitImpl implements Homekit {
+
+    private final HomekitSettings settings = new HomekitSettings();
+    private HomekitServer homekit;
+    private HomekitRoot bridge;
+    private StorageService storageService;
+    private final HomekitChangeListener homekitRegistry = new HomekitChangeListener();
```

Comment:
why do you call the variable registry when the class is a change listener...? Feels a bit weird.